### PR TITLE
feat(android-settings): update how we suppress the first time dialog

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

Currently, our AI-Android e2e tests dismiss the first time dialog (the telemetry opt-in) by programmatically interacting with it. This is not the way we do it on AI-Web, which uses a "window"-exposed controller.

In this PR:
- Update our app controller to expose a wait to interact with the user configuration controller on the browser window
- Update our e2e tests so they use the new way to suppress the first time dialog
- Add an e2e test to verify the controller properly update the store and updates the settings panel ui

#### Pull request checklist
- [x] Addresses an existing issue: completes WI # 1687239
- [x] Ran `yarn fastpass`
- [ ] Added/updated relevant unit test(s) (and ran `yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). Check workflow guide at: `<rootDir>/docs/workflow.md`
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/electron/common/view-controllers/app-controller.ts [hunk 1]
```diff
@@ -71,4 +68,10 @@ export class AppController {
             } class high-contrast-theme`,
         );
     }
+
+    public async setTelemetryState(enableTelemetry: boolean): Promise<void> {
+        await this.app.webContents.executeJavaScript(
+            `window.insightsUserConfiguration.setTelemetryState(${enableTelemetry})`,
```
Comment (dbjorge) [inline src/tests/electron/common/view-controllers/app-controller.ts]: This might be happening too early, since it's going before we wait for a view to be visible; maybe better would be for it to involve some sort of `waitUntil` that verifies `window.insightsUserConfiguration` is defined first?

File: src/tests/electron/common/dismiss-telemetry-opt-in-dialog.ts [hunk 2]
```diff
@@ -1,16 +0,0 @@
...
-// Licensed under the MIT License.
-import { Application } from 'spectron';
-import { DEFAULT_ELECTRON_TEST_TIMEOUT_MS } from 'tests/electron/setup/timeouts';
-import { popupPageElementIdentifiers } from 'tests/end-to-end/common/element-identifiers/popup-page-element-identifiers';
-import { Client } from 'webdriverio';
-
-export async function dismissTelemetryOptInDialog(app: Application): Promise<void> {
-    const webDriverClient: Client<void> = app.client;
-
-    await webDriverClient.waitForVisible(
```
Comment (dbjorge) [inline src/tests/electron/common/dismiss-telemetry-opt-in-dialog.ts]: It would be nice to move this to a test that verifies it has the expected "suppress telemetry" behavior rather than removing it outright, but if you'd rather leave that for a separate PR that's understandable

Comment (smoralesd) [reply src/tests/electron/common/dismiss-telemetry-opt-in-dialog.ts]: I though about that to, like: "uh... we don't have an actual test to check the first time dialog shows up the first time and that we can dismiss it properly". I'll add a test on a separated PR.

File: src/tests/electron/common/view-controllers/app-controller.ts (hunk 1 above)
Comment (smoralesd) [reply src/tests/electron/common/view-controllers/app-controller.ts]: that's a good point. I'll work on it.

Comment (smoralesd) [reply src/tests/electron/common/view-controllers/app-controller.ts]: Updated. Is this close to what you had in mind?

Comment (dbjorge) [reply src/tests/electron/common/view-controllers/app-controller.ts]: Yes, this seems about right to me!
